seqr/views/utils/project_context_utils.py

```diff
@@ -0,0 +1,140 @@
+
+
+def get_projects_child_entities(projects, user, is_analyst=None):
+    project_guid = projects[0].guid if len(projects) == 1 else None
+    has_case_review_perm = has_case_review_permissions(projects[0], user)
+    if is_analyst is None:
+        is_analyst = user_is_analyst(user)
+
+    response = _fetch_child_entities(projects, project_guid, user, is_analyst, has_case_review_perm)
+
+    _add_tag_types(response['projectsByGuid'], project_guid)
+
+    return response
+
+def _fetch_child_entities(projects, project_guid, user, is_analyst, has_case_review_perm):
+    projects_by_guid = {p['projectGuid']: p for p in get_json_for_projects(projects, user)}
+
+    family_models = Family.objects.filter(project__in=projects)
+    families = _get_json_for_families(
+        family_models, user, project_guid=project_guid, skip_nested=True,
+        is_analyst=is_analyst, has_case_review_perm=has_case_review_perm)
+
+    individual_models = Individual.objects.filter(family__in=family_models)
+    individuals = _get_json_for_individuals(
+        individual_models, user=user, project_guid=project_guid, add_hpo_details=True, skip_nested=True,
+        is_analyst=is_analyst, has_case_review_perm=has_case_review_perm)
+
+    sample_models = Sample.objects.filter(individual__in=individual_models)
+    samples = get_json_for_samples(sample_models, project_guid=project_guid, skip_nested=True)
+
+    igv_sample_models = IgvSample.objects.filter(individual__in=individual_models)
+    igv_samples = get_json_for_samples(igv_sample_models, project_guid=project_guid, skip_nested=True)
+
+    analysis_group_models = AnalysisGroup.objects.filter(project__in=projects)
+    analysis_groups = get_json_for_analysis_groups(analysis_group_models, project_guid=project_guid, skip_nested=True)
+
+    locus_lists_models = LocusList.objects.filter(projects__in=projects)
+    locus_lists_by_guid = {
+        ll['locusListGuid']: ll for ll in get_json_for_locus_lists(locus_lists_models, user, is_analyst=is_analyst)}
+
+    response = {
+        'projectsByGuid': projects_by_guid,
+        'familiesByGuid': {f['familyGuid']: f for f in families},
+        'individualsByGuid': {i['individualGuid']: i for i in individuals},
+        'samplesByGuid': {s['sampleGuid']: s for s in samples},
+        'igvSamplesByGuid': {s['sampleGuid']: s for s in igv_samples},
+        'locusListsByGuid': locus_lists_by_guid,
+        'analysisGroupsByGuid': {ag['analysisGroupGuid']: ag for ag in analysis_groups},
+    }
+
+    if project_guid:
+        response['projectsByGuid'][project_guid]['locusListGuids'] = list(response['locusListsByGuid'].keys())
+    else:
+        _add_parent_ids(response, projects, family_models, individual_models, locus_lists_models)
+
+    _add_child_ids(response)
+
+    return response
+
+def _add_parent_ids(response, projects, family_models, individual_models, locus_lists_models):
```

Comment:
I'm curious that how does it relate to `parent`?

Reply:
The parent models. As in, a project is a parent of a family, a family is a parent of an individual, etc

Reply:
Okay. I thought the parents in the family pedigree.